include/fastdds/dds/topic/TopicDataType.hpp

```diff
@@ -107,10 +108,9 @@ class TopicDataType
             fastrtps::rtps::SerializedPayload_t* payload,
             DataRepresentationId_t data_representation)
     {
-        static_cast<void>(data);
-        static_cast<void>(payload);
+
```

Comment:
We should take the opportunity to create a cpp for this with the implementation. Also, we need to deprecate the other overload and mark it as to be removed by 3.0.0.

Reply:
- Cpp added in [a10299c](https://github.com/eProsima/Fast-DDS/pull/4131/commits/a10299c8f129123222c132bfc9a9370f9d84054a)

- Deprecation note in [685e4fc](https://github.com/eProsima/Fast-DDS/pull/4131/commits/685e4fc22291037e3e45dd21b482cb068ccfe209)